### PR TITLE
SQLA 2.0 compatibility tweaks

### PR DESCRIPTION
**Pull request by marcospri**
More SQLA related changes that can be made in 1.4 in preparation for the 2.0 upgrade.

### CODE REVIEW

File: h/db/types.py [hunk 1]
```diff
@@ -29,6 +29,10 @@ class URLSafeUUID(types.TypeDecorator):  # pylint:disable=abstract-method
     impl = postgresql.UUID
     cache_ok = True
 
+    def __init__(self):
+        # We handle the UUID conversion, explicitly not use as_uuid
+        super().__init__(as_uuid=False)
```
Comment (marcospri) [inline h/db/types.py:34]: Default changes in 2.0

File: h/db/types.py [hunk 2]
```diff
@@ -136,6 +140,7 @@ class AnnotationSelectorJSONB(types.TypeDecorator):  # pylint:disable=abstract-m
     """
 
     impl = postgresql.JSONB
+    cache_ok = False
```
Comment (marcospri) [inline h/db/types.py:143]: 2.0 needs this defined, (ie not None). False is the value now effectively

File: h/search/index.py [hunk 3]
```diff
@@ -129,7 +129,7 @@ def _eager_loaded_annotations(session):
         ),
         subqueryload(models.Annotation.document).subqueryload(models.Document.meta),
         subqueryload(models.Annotation.moderation),
-        subqueryload(models.Annotation.thread).load_only("id"),
+        subqueryload(models.Annotation.thread).load_only(models.Annotation.id),
```
Comment (marcospri) [inline h/search/index.py:132]: No string columns

File: tests/unit/h/models/user_identity_test.py [hunk 4]
```diff
@@ -123,43 +123,24 @@ def test_different_users_can_have_the_same_provider_id_from_different_providers(
 
         db_session.flush()
 
-    def test_removing_a_user_identity_from_a_user_deletes_the_user_identity_from_the_db(
```
Comment (marcospri) [inline tests/unit/h/models/user_identity_test.py:126]: Just keep the assertion of this test and added to the next.

File: tests/unit/h/models/user_identity_test.py [hunk 5]
```diff
@@ -123,43 +123,24 @@ def test_different_users_can_have_the_same_provider_id_from_different_providers(
...
-            provider="provider", provider_unique_id="3", user=user
-        )
-
-        user.identities.remove(user_identity)
-
-        assert user_identity not in db_session.query(models.UserIdentity).all()
-
+    @pytest.mark.usefixtures("user_identities")
     def test_deleting_a_user_identity_removes_it_from_its_user(self, db_session, user):
-        # Add a couple of noise UserIdentity's. These should not be removed
```
Comment (marcospri) [inline tests/unit/h/models/user_identity_test.py:143]: Moved to fixture

**Review by seanh — APPROVED** [review]

**Pull request merged**